plugin/log/setup.go

```diff
@@ -51,45 +54,57 @@ func logParse(c *caddy.Controller) ([]Rule, error) {
 		} else {
 			// Name scope, and maybe a format specified
 			format := DefaultLogFormat
+			last := len(args)
 
-			switch args[1] {
-			case "{common}":
-				format = CommonLogFormat
-			case "{combined}":
-				format = CombinedLogFormat
-			default:
-				format = args[1]
+			if strings.Contains(args[last-1], "{") {
+				last--
+				switch args[last] {
+				case "{common}":
+					format = CommonLogFormat
+				case "{combined}":
+					format = CombinedLogFormat
+				default:
+					format = args[last]
+				}
 			}
 
-			rules = append(rules, Rule{
-				NameScope: dns.Fqdn(args[0]),
-				Format:    format,
-				Class:     make(map[response.Class]struct{}),
-			})
+			args = args[:last]
+			for _, str := range args {
+				rules = append(rules, Rule{
+					NameScope: dns.Fqdn(str),
+					Format:    format,
+					Class:     make(map[response.Class]struct{}),
+				})
+			}
 		}
 
 		// Class refinements in an extra block.
+		classes := make(map[response.Class]struct{})
 		for c.NextBlock() {
 			switch c.Val() {
 			// class followed by combinations of all, denial, error and success.
 			case "class":
-				classes := c.RemainingArgs()
-				if len(classes) == 0 {
+				classesArgs := c.RemainingArgs()
+				if len(classesArgs) == 0 {
 					return nil, c.ArgErr()
 				}
-				for _, c := range classes {
+				for _, c := range classesArgs {
 					cls, err := response.ClassFromString(c)
 					if err != nil {
 						return nil, err
 					}
-					rules[len(rules)-1].Class[cls] = struct{}{}
+					classes[cls] = struct{}{}
 				}
 			default:
 				return nil, c.ArgErr()
 			}
 		}
-		if len(rules[len(rules)-1].Class) == 0 {
-			rules[len(rules)-1].Class[response.All] = struct{}{}
+		if len(classes) == 0 {
+			classes[response.All] = struct{}{}
+		}
+
+		for i := len(rules) - 1; i >= length; i -= 1 {
+			rules[i].Class = classes
```

Comment:
the class map is now shared between all rules? Might make sense, but should probably be done separately?

Reply:
ah yes I see. Let's not make this change right now though.